doc/RELEASE.adoc

```diff
@@ -0,0 +1,132 @@
+The right group of people should be involved and sign off on the tests and checks.
+5. If no release blocking issue is found, tag `x.y.z.0` and upload the packages to https://github.com/input-output-hk/cardano-haskell-packages[CHaP].
+- If issues are found, create a release branch `release/x.y.z`, fix the issues on master, backport the fixes to `release/x.y.z`, tag `x.y.z.0-rc2`, and go to step 4.
+- Why not just fix the issues on master and tag `x.y.z.0-rc2` from master?
+It is desirable to minimize the amount of change between `rc1` and `rc2`, because it may reduce the tests and checks that need to be performed against `rc2`.
+For instance, if `plutus-ledger-api` is the only package changed, there is no need to re-run tests on `plutus-core` or `plutus-tx`.
+Another example is if a security audit is done on `rc1`, and the changes in `rc2` do not modify the audited code, then the audit does not need to be re-done.
+
+6. After the release, update version numbers in cabal files to `x.y.z+1.0`.
+- This may not be the actual version of the next release - it could be `x+1.0.0`.
+We don't need to be precise here; we just need a version number higher than the latest release.
+
+=== Patch Releases
+
+Suppose we are releasing version `x.y.z.w`.
+
+1. If it is the first time we are making a patch release for version `x.y.z` (i.e., `x.y.z.0` -> `x.y.z.1`), a release branch may not exist for `x.y`.
+If so, create branch `release/x.y.z` from the `x.y.z.0` tag.
+- We create release branches lazily, because we do not expect to make many patch releases.
+2. Backport the needed fixes from master to `release/x.y.z`.
+3. Tag `x.y.z.w-rc1` on the release branch.
+4. Run all tests and checks.
+5. If no release blocking issue is found, tag version `x.y.z.w` and upload the packages to CHaP.
+- If issues are found, fix them on master, backport the fixes to `release/x.y.z`, and go to step 3.
+
+== Backwards Compatibility with Cardano API
+
+It is a good idea to avoid breaking the latest version of Cardano API in a new Plutus release.
+This makes it easy for downstream projects to update Plutus version without needing a new Cardano API release.
+
+For example, suppose we make some improvements to `plutus-tx-plugin` and make a new major release.
+Since all packages are released in sync, we also make a new major release for `plutus-core`.
+Although it is a major release, we should avoid making changes that is incompatible with the latest version of Cardano API.
+Otherwise, downstream projects such as Plutus Tools won't be able to use the new Plutus version and take advantage of the plugin improvements, until a new Cardano API version is published.
+
+To do so, rather than making changes to the Plutus API that breaks Cardano API (e.g., changing the type of a function), we can temporarily keep both the old Plutus API and the new Plutus API, until a new Cardano API version is released that no longer depends on the old Plutus API.
```

Comment:
I'm still a bit confused by this. So when _can_ we break `cardano-api`? Apparently not on a major version so... never?

Reply:
The idea is not to break the _latest_ version of `cardano-api`. I'll clarify that this is a suggestion rather than a hard rule.